src/ralph/backends/http/async_lrs.py

```diff
@@ -150,10 +151,10 @@ def read(  # pylint: disable=too-many-arguments
             fragment="",
         ).geturl()
 
-        def fetch_statements(params: dict):
+        async def fetch_statements(params: dict):
```

Comment:
Maybe the client can be instanciated once in the `read` method and injected in the `fetch_statements` function to avoid having a client instance per request. Hat tip to @cyrillay for the idea that I tested lately.

Reply:
Good point. I moved the client instantiation up (before `while True`).